### PR TITLE
added turret subsystem

### PR DESCRIPTION
**Pull request by lydiaxing**

### CODE REVIEW

**Comment by lydiaxing** [comment]
merge conflicts fixed

File: arduino/vision_runner/vision_runner.ino [hunk 1]
```diff
@@ -15,13 +15,13 @@ double FOV_X = 75; //horizontal Field of View in degrees
 double FOV_Y = 47; //vertical Field of View in degrees
```
Comment (MiloDavis) [inline arduino/vision_runner/vision_runner.ino:15]: We need to discuss this whole file.  See gear vision PR

File: src/com/nutrons/steamworks/Shooter.java [hunk 2]
```diff
@@ -14,7 +17,7 @@ public Shooter(LoopSpeedController shooterController) {
 
   @Override
   public void registerSubscriptions() {
-    shooterController.setPID(0.05, 0.0, 0.33, 0.035);
-    shooterController.setSetpoint(2950.0);
+    Flowable<ControllerEvent> source = toFlow(() -> new LoopPropertiesEvent(2950.0, 0.05, 0.0, 0.33, 0.035));
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Shooter.java]: No magic numbers

Comment (MiloDavis) [reply src/com/nutrons/steamworks/Shooter.java]: Just use Flowable.just

File: src/com/nutrons/steamworks/Shooter.java [hunk 3]
```diff
@@ -14,7 +17,7 @@ public Shooter(LoopSpeedController shooterController) {
...
+    source.mergeWith(toFlow(() -> new LoopModeEvent(ControlMode.LOOP_SPEED))).subscribe(shooterController);
   }
-}
+}
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Shooter.java]: Newline at end of file

File: src/com/nutrons/steamworks/Turret.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import io.reactivex.Flowable;
+
+import static com.nutrons.framework.util.FlowOperators.toFlow;
+
+public class Turret implements Subsystem {
+  private final Flowable<Double> angle;
+  private final LoopSpeedController hoodMaster;
+
+  private volatile double motorRotation;
+  private static final double MOTOR_ROTATIONS_TO_TURRET_ROTATIONS = 104 / 22;
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Turret.java]: Cast one of these to double.  You're probably getting this rounded off

File: src/com/nutrons/steamworks/Turret.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  public Turret(Flowable<Double> angle, LoopSpeedController master) {
+    this.angle = angle;
+    this.hoodMaster = master;
+    this.angle.map(x -> x * MOTOR_ROTATIONS_TO_TURRET_ROTATIONS / 360.0).subscribe(x -> this.motorRotation = x);
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Turret.java]: This stream should link up with the controller events.  You shouldn't have them communicate via side effect.  Come talk to me about this today.

File: src/com/nutrons/steamworks/Vision.java [hunk 6]
```diff
@@ -8,12 +8,15 @@
   private Flowable<Double[]> dataStreamDouble;
   private Flowable<Double> angle;
   private Flowable<Double> distance;
+  private static final String DUMMY_VALUE = "-1000:-1000"; //Arduino sends -1000.0 over serial when it doesn't see anything, to prevent
```
Comment (MiloDavis) [inline src/com/nutrons/steamworks/Vision.java]: Move comment over line

**Review by MiloDavis — DISMISSED** [review]

File: src/com/nutrons/steamworks/Turret.java (hunk 5 above)
Comment (lydiaxing) [reply src/com/nutrons/steamworks/Turret.java]: dont' save to MotorRotation, just map straight to map(Events.pid(offset + x, p, i , d, f);

File: arduino/vision_runner/vision_runner.ino (hunk 1 above)
Comment (lydiaxing) [reply arduino/vision_runner/vision_runner.ino:15]: fixed under gear vision pr

**Review by kreeve — APPROVED** [review]

**Pull request merged**